### ISSUE TITLE
@global and window object  

### ISSUE THREAD

**Issue by nolochemical** [ractivejs/ractive#3140]
Afternoon,

I hoping to gain clarity  `@global vs window `  as far as I understand they are one in the same (for browser clients). I'm hoping to do something like this with a template

`{{ @global.setTimeout( ()=>{ alert('hello world') }, 500) }}`

../js/src/parse/converters/mustache/readInterpolator.js :: ln 25
Ph {name: "ParseError", message: "Expected a JavaScript expression at

**Comment by ghost** [comment]
Ractive has [limited expression support inside mustaches](https://ractive.js.org/api/#expressions). You're creating a function (in this case, an arrow function) inside the mustache, which is not allowed.

If you want to run a function on render, use the [`onrender`](https://ractive.js.org/api/#onrender) lifecycle function instead.

**Comment by evs-chris** [comment]
You can also run a data function in an interpolator e.g. `{{ ~/doTheTimeoutAlert() }}` with `data: { doTheTimeoutAlert() { setTimeout( ... ); } }` in your init options if you need to wait on the actual interpolator render in case it's in a conditional or something.

But yes, `@global` _is_ the `window` object in a browser and `global` in node. You just can't create functions in the template at this point, and if support for arrows does get added (probably), it will likely only be for the expression form (without a body block) e.g. `() => alert('...')` vs `() => { ... }` because the template has no concept of statements.

**Comment by nolochemical** [comment]
@evs-chris Yes, it was a conditional thanks for the clarity 
@fskreuz Thanks I will take a look

**Comment by ghost** [comment]
TL;DR: `@global` is `window`, limited expression support, statements not allowed. I think I can close this one. 😁
